src/allmydata/client.py

```diff
@@ -179,42 +203,86 @@ def _sequencer(self):
         nonce = _make_secret().strip()
         return seqnum, nonce
 
-    def init_introducer_clients(self):
-        self.introducer_furls = []
+    def load_connections_from_yaml(self, furl):
+        connections_filepath = FilePath(os.path.join(self.basedir, "private", "connections.yaml"))
+        if connections_filepath.exists():
+            exists = True
+            with connections_filepath.open() as f:
+                connections = yaml.load(f)
+                f.close()
+        else:
+            exists = False
+            connections = { 'introducers' : {},
+                            'servers' : {},
+                            'transport_plugins' : {
+                                'tcp' : {
+                                    'handler_module' : 'foolscap.connection_plugins',
+                                    'handler_name': 'DefaultTCP',
+                                    'parameters' : {}
+                                },
+                            },
+                            }
+            new_connections = connections.copy()
+            new_connections['introducers'][u'default'] = {}
+            new_connections['introducers']['default']['furl'] = furl
+            connections_filepath.setContent(yaml.dump(new_connections))
+        return connections, exists
+
+    def load_connections(self):
+        """
+        Load the connections.yaml file if it exists, otherwise
+        create a default configuration. Abort startup and report
+        an error to the user if the tahoe.cfg contains an introducer
+        FURL which is also found in the connections.yaml.
+        """
+        # read introducer from tahoe.cfg and abort + error an introducer furl is specified
+        # which is also found in our connections.yaml
```

Comment:
this comment is wrong, there should be an error if a furk is specified in `tahoe.cfg` and that furl is *not* in connections.yaml (if it exists)